gdnative-core/src/core_types/mod.rs

```diff
@@ -5,34 +5,36 @@
 //!
 //! godot-rust provides optional serialization support for many core types.  Enable the feature `serde` to make use of it.
 
-mod geom;
-
-mod access;
 mod color;
-mod dictionary;
 mod error;
 mod node_path;
 mod pool_array;
 mod rid;
-mod string;
-mod variant;
-mod variant_array;
 mod vector2;
 mod vector3;
 
-pub use access::*;
-pub use color::*;
-pub use dictionary::*;
+pub mod access;
+pub mod array;
+pub mod dictionary;
+pub mod geom;
+pub mod string;
+pub mod variant;
+
+pub use array::VariantArray;
+pub use color::Color;
+pub use dictionary::Dictionary;
 pub use error::{GodotError, GodotResult};
-pub use geom::*;
-pub use node_path::*;
-pub use pool_array::*;
-pub use rid::*;
-pub use string::*;
-pub use variant::*;
-pub use variant_array::*;
-pub use vector2::*;
-pub use vector3::*;
+pub use geom::{Aabb, Basis, Margin, MarginError, Plane, Quat, Rect2, Transform, Transform2D};
+pub use node_path::NodePath;
+pub use pool_array::{PoolArray, PoolElement};
+pub use rid::Rid;
+pub use string::{GodotString, StringName};
+pub use variant::{
+    CoerceFromVariant, FromVariant, FromVariantError, OwnedToVariant, ToVariant, ToVariantEq,
+    Variant, VariantType,
+};
+pub use vector2::Vector2;
+pub use vector3::{Axis, Vector3};
```

Comment:
Maybe general question about these:

Should it not be up to the module to define which symbols are `pub` and which not? This approach spreads the visibility decision across two places, which I found harder to maintain in the past (see e.g. bugs like https://github.com/PoignardAzur/venial/pull/30).

Reply:
That's true, but the flip side is that there exist types that should be `pub`, but should not appear in the higher-level module. The motivation behind these changes, `Iter` / `IntoIter` and `Read` / `Write`, are prime examples of these. Some of the error detail types from `variant` too. The only way I see to model this without using selective exports at the parent level would be something like:

```rust
// inner_module.rs
pub(super) mod exports {
    pub use super::{Foo, Bar, Baz};
}

// mod.rs
pub mod inner_module;
pub use inner_module::exports::*;
```

Which does not exactly feel like a very convenient pattern to me. In either case there's still a separate list to maintain apart from the visibility modifier that one can forget to update.